sequence_task_2/student/include/impl.h

```diff
@@ -14,4 +14,6 @@
  * @note See https://en.cppreference.com/w/cpp/algorithm/rotate
  */
 template<class T, class BI>
-void rotate(std::list<T>& list, BI begin, BI new_begin, BI end);
+void rotate(std::list<T>& list, BI begin, BI new_begin, BI end) {
+    list.splice(end, list, begin, new_begin);
```

Comment:
what if new_begin is less than begin? so we'll get the invalid range (begin, new_begin)